command/agent/agent.go

```diff
@@ -127,28 +130,64 @@ func Create(config *Config, logOutput io.Writer) (*Agent, error) {
 		logOutput = os.Stderr
 	}
 
-	// Validate the config
-	if config.Datacenter == "" {
-		return nil, fmt.Errorf("Must configure a Datacenter")
-	}
-	if config.DataDir == "" && !config.DevMode {
-		return nil, fmt.Errorf("Must configure a DataDir")
-	}
-
-	// Try to get an advertise address
+	var bindIPs []string
 	if config.AdvertiseAddr != "" {
-		if ip := net.ParseIP(config.AdvertiseAddr); ip == nil {
-			return nil, fmt.Errorf("Failed to parse advertise address: %v", config.AdvertiseAddr)
+		// Parse config.AdvertiseAddr to resolve an IP address
+		out, err := template.Parse(config.AdvertiseAddr)
```

Comment:
I think you need to make sure the result of the parse only has one address and then set `config.AdvertiseAddr` to that result, since you won't parse this template again.